### PR TITLE
[10.x] Add getQualifiedTableName Method to Eloquent Model

### PR DESCRIPTION
**Pull request by mrfelipemartins**
This PR introduces a new method,`getQualifiedTableName`, to the `Illuminate\Database\Eloquent\Model` class. This method aims to simplify the process of referencing fully qualified table names (including the connection name, when applicable) in multi-database setups. It concatenates the connection name and the table name. This is especially useful when performing joins across multiple connections.

The `getQualifiedTableName` method checks if the model's connection name is set; if so, it returns the connection name and table name concatenated with a dot (.). If no connection name is set, it returns just the table name, preserving the expected behavior for models using the default database connection.

### CODE REVIEW

File: tests/Foundation/FoundationInteractsWithDatabaseTest.php [hunk 1]
```diff
@@ -30,6 +30,8 @@ class FoundationInteractsWithDatabaseTest extends TestCase
     protected function setUp(): void
     {
         $this->connection = m::mock(Connection::class);
+
+        $this->connection->shouldReceive('getName')->andReturn('test_connection');
```
Comment (driesvints) [inline tests/Foundation/FoundationInteractsWithDatabaseTest.php:34]: Why did you need to add this to the `setUp` method? Does this PR changes the behaviour of all method calls in Eloquent?

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.
